### DISCUSSION TITLE
Google Compute Engine Image Versions

### DISCUSSION THREAD

**Discussion by TJM** [renovatebot/renovate#17489] [Suggest an Idea]
We are using Google Cloud Platform, but I couldn't find any examples for it, nor AWS/Azure/etc. Perhaps I am not searching on the right terms. When we create a "VM" in Google Cloud Platform (GCP) in Google Compute Engine (GCE), we specify something like:

```
source_image = "debian-cloud/debian-10-buster-v20220822"
```

... in our terraform code. We could use a "data" resource to retrieve the most recent version, but then OS updates happen while making some other, unrelated change, which is bad. We want the updates to be "intentional" updates. I figured, for sure, Renovate would be able to support this, but either I can't find it in the docs or compute images are not supported. The package manager data could be obtained using a command, `gcloud compute images list`, (possibly with an optional  `--project projectID`) or the equivalent API call. 

SO, questions:

- Does the capability already exist to look for these updates (and I am just poor at searching)?
- Assuming it does not exist, is there a way to add additional package sources? (this seems like it may be possible in the docs there are notes about "sidecar containers")
- Has anyone already done something like this?  'Cause even if we had to "modify" it to suit our needs, it still sounds better than re-inventing this wheel.

Thanks in advance,
Tommy

**Comment by rarkins** [comment]
It looks like this would need a new Renovate "datasource" added, unless this is actually just Docker images underneath?

**Reply by TJM** [reply]
That would be nice, but they are not docker images, this is the VM (compute) images. I messed around with `gcloud compute images list` for a while yesterday, but at least in the default output (--format=json) do not seem to separate image *name* from *version* :(

**Reply by viceice** [reply]
renovate can't use gcloud cli anyways, we need a http API to get those information

**Comment by vidbregar** [comment]
I was looking for something similar recently. As of now, the best approach seems to be building a custom datasource (it’s actually pretty straightforward). This gives you the flexibility to upgrade almost anything.

I’ve documented the process here: [Automating Google Cloud Image Upgrades with a Custom Renovate Datasource](https://www.vidbregar.com/blog/renovate-gcp-images)

**Reply by jamietanna** [reply]
Thanks for sharing! Getting this as a first-class feature (as a Datasource that Renovate supports natively) would be great, if anyone is interested in contributing!
